File: android/model/src/main/kotlin/com/dgsd/hackernews/model/Story.kt
package com.dgsd.hackernews.model

import java.util.concurrent.TimeUnit

public data class Story(
        val id: Long = -1,
        val time: Long = -1,
        val type: Story.Type = Story.Type.STORY,
        val parentId: Long = -1,
        val author: String? = null,
        val title: String? = null,
        val text: String? = null,
        val url: String? = null,
        val commentCount: Int = -1,
        val score: Int = -1,
        val pollAnswers: List<Long> = emptyList(),
        val commentIds: List<Long> = emptyList(),
        val comments: List<Comment> = emptyList(),
        val dateRetrieved: Long = TimeUnit.MILLISECONDS.toSeconds(System.currentTimeMillis())) {

    public enum class Type {
        STORY, JOB, POLL, POLL_ANSWER
    }

}

File: ios/hackernews/models/Story.swift
//
//  Story.swift
//  hackernews
//
//  Created by Daniel Grech on 12/10/2015.
//  Copyright © 2015 DGSD. All rights reserved.
//

import Foundation

enum StoryType {
    case Story
    case Job
    case Poll
    case Poll_Answer
}

struct Story {
    
    var id: Int64 = -1
    var time: Int64 = -1
    var type: StoryType = StoryType.Story
    var parentId: Int64 = -1
    var author: String? = nil
    var title: String? = nil
    var text: String? = nil
    var url: String? = nil
    var commentCount: Int32 = -1
    var score: Int32 = -1
    var pollAnswers: [Int64] = []
    var commentIds: [Int64] = []
    var comments: [Comment] = []
    var deleted: Bool
    var dead: Bool
    var dateRetrieved: Int64 = NSDate.currentTimeInMillis()
    
    func hasComments() -> Bool {
        return !commentIds.isEmpty
    }
    
    func hasCommentsToLoad() -> Bool {
        return hasComments() && comments.isEmpty
    }
    
}